torch/utils/data/datapipes/iter/callable.py

```
@@ -227,3 +228,60 @@ def __init__(
                 # TODO(VitalyFedyunin): Validate passed dictionary
                 collate_fn = functools.partial(_collate_helper, conversion)
                 super().__init__(datapipe, fn=collate_fn)
+
+@functional_datapipe('drop')
+class DropperIterDataPipe(IterDataPipe):
+    r"""
+    Drop columns/elements in input DataPipe via its indices (functional name: ``drop``).
+
+    Args:
+        datapipe: MapDataPipe with columns to be dropped
+        indices: a single column index to be dropped or a list of indices
+
+    Example:
+        >>> from torchdata.datapipes.iter import IterableWrapper, ZipperMapDataPipe
+        >>> dp1 = IterableWrapper(range(5))
+        >>> dp2 = IterableWrapper(range(10, 15))
+        >>> dp = dp1.zip(dp2)
+        >>> list(dp)
+        [(0, 10), (1, 11), (2, 12), (3, 13), (4, 14)]
+        >>> drop_dp = dp.drop(1)
+        >>> list(drop_dp)
+        [(0), (1), (2), (3), (4)]
+    """
+    datapipe: IterDataPipe
+
+    def __init__(self,
+                 datapipe: IterDataPipe,
+                 indices: Union[int, List],
+                 ) -> None:
+        super().__init__()
+        self.datapipe = datapipe
+        if isinstance(indices, list):
+            self.indices = set(indices)
+        else:
+            self.indices = {indices}
+
+    def __iter__(self) -> T_co:
+        for old_item in self.datapipe:
+            if isinstance(old_item, tuple):
+                new_item = tuple(x for i, x in enumerate(old_item) if i not in self.indices)
+            elif isinstance(old_item, list):
+                new_item = [x for i, x in enumerate(old_item) if i not in self.indices]
+            elif isinstance(old_item, dict):
+                new_item = {k: v for (k, v) in old_item.items() if k not in self.indices}    
```

Should we raise warning when any of `indices` is not presented in `old_item`?